streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -413,6 +417,10 @@ private StreamsGraphNode getKeyChangingParentNode(final StreamsGraphNode reparti
         return null;
     }
 
+    private String getFirstRepartitionTopicName(final Collection<OptimizableRepartitionNode> repartitionNodes) {
+        return repartitionNodes.iterator().next().repartitionTopic();
```

Comment:
Did we want to give priority to the explicitly named repartition topics?

Reply:
I had the same thought, but so far we've only discussed grabbing the first repartition topic name. 
 I'm inclined to leave as is because 1) users don't care about the name as much as it doesn't change and break the topology and 2) IMHO will add some complexity without a significant benefit

Reply:
Ack.

It's probably also likely that the user who names some repartition topics names them all; another reason the extra complexity wouldn't buy anything.